src/OrchardCore/Orchard.Localization.Abstractions/CultureDictionary.cs

```diff
@@ -6,44 +6,41 @@ namespace Orchard.Localization.Abstractions
 {
     public class CultureDictionary
     {
-        private IDictionary<string, string[]> _translations;
-
         public string CultureName { get; private set; }
 
-        public Func<int, int> PluralRule { get; private set; }
+        public PluralRuleDelegate PluralRule { get; private set; }
 
         public string this[string key] => this[key, null];
 
         public string this[string key, int? count]
         {
             get
             {
-                if(key == null)
+                if (key == null)
                 {
                     throw new ArgumentNullException(nameof(key));
                 }
 
-                var translations = _translations.ContainsKey(key) ? _translations[key] : null;
-                if (translations == null)
+                if (!Translations.TryGetValue(key, out string[] translations))
                 {
                     return null;
                 }
 
                 var pluralForm = count.HasValue ? PluralRule(count.Value) : 0;
-                if (pluralForm < translations.Length)
+                if (pluralForm >= translations.Length)
                 {
-                    return translations[pluralForm];
+                    throw new PluralFormNotFoundException($"Plural form '{pluralForm}' doesn't exist for the key '{key}' in the '{CultureName}' culture.");
```

Comment:
Was I that evil to ask you to throw an exception? Looking at it maybe a warning log would be sufficient. Or it would break incomplete localization files. And return the standard form if no localization can be found.